### PR TITLE
Add edge caching logic to series pages

### PR DESCRIPTION
**Pull request by benhalpern**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [x] Optimization
- [ ] Documentation Update

## Description

I noticed these pages were a little slow. Adding edge caching headers and small optimization.

Also skipping displaying collections with no articles.

### CODE REVIEW

File: app/controllers/collections_controller.rb [hunk 1]
```diff
@@ -1,12 +1,17 @@
 class CollectionsController < ApplicationController
+  before_action :set_cache_control_headers, only: %i[index show]
+
   def index
     @user = User.find_by!(username: params[:username])
-    @collections = @user.collections.order(created_at: :desc)
+    @collections = @user.collections.includes(:articles).order(created_at: :desc)
+    @articles = Article.published.where(collection_id: @collections.pluck(:id))
```
Comment (citizen428) [inline app/controllers/collections_controller.rb:7]: **suggestion (non-blocking):** Based on @rhymes recent PR, I think you may be able to use `@collections.ids` here.

File: app/views/collections/index.html.erb [hunk 2]
```diff
@@ -12,6 +12,7 @@
   <div class="grid gap-2 m:gap-4 l:gap-6 m:grid-cols-2 l:grid-cols-3 px-2 m:px-0">
     <% if @collections.present? %>
       <% @collections.each do |collection| %>
+        <% next if collection.articles.published.empty? %>
```
Comment (citizen428) [inline app/views/collections/index.html.erb:15]: **thought (non-blocking):** I'm wondering if these could already be filtered out in the controller, i.e. should `@collections` only include collections with at least one published article?

File: app/controllers/collections_controller.rb (hunk 1 above)
Comment (rhymes) [reply app/controllers/collections_controller.rb:7]: Yeah, theoretically rubocop should have flagged this

Comment (rhymes) [reply app/controllers/collections_controller.rb:7]: Shouldn't we use `Collection.table_key` here? The collections are going to be updated when the article is updated:

https://github.com/forem/forem/blob/6ac637b9ef4c74e09201065abe662e2d1e9e8952/app/models/article.rb#L90

https://github.com/forem/forem/blob/6ac637b9ef4c74e09201065abe662e2d1e9e8952/app/models/article.rb#L672-L674

File: app/controllers/collections_controller.rb [hunk 3]
```diff
@@ -1,12 +1,17 @@
...
+    set_surrogate_key_header Article.table_key, @articles.map(&:record_key)
   end
 
   def show
     @collection = Collection.find(params[:id])
     @user = @collection.user
     @articles = @collection.articles.published.order(Arel.sql("COALESCE(crossposted_at, published_at) ASC"))
+    set_surrogate_key_header Article.table_key, @articles.map(&:record_key)
```
Comment (rhymes) [inline app/controllers/collections_controller.rb:15]: Here as well, I think we can manage by just using the collection key

**Comment by msarit** [comment]
@benhalpern Hey Ben! 👋🏾  Just drawing your attention back to this PR re: comments 😄

**Comment by mstruve** [comment]
Hey! Since this still has some failing specs and conflicting files and has not been active for a while I am going to close it. I think if we run into issues in the future with these pages we can revisit this approach.

**Pull request closed without merging**